extensions/interactions/MathExpressionInput/MathExpressionInput.js

```diff
@@ -30,13 +30,125 @@ oppia.directive('oppiaInteractiveMathExpressionInput', [
       templateUrl: 'interaction/MathExpressionInput',
       controller: [
         '$scope', '$attrs', '$timeout', '$element', 'LABEL_FOR_CLEARING_FOCUS',
-        function($scope, $attrs, $timeout, $element, LABEL_FOR_CLEARING_FOCUS) {
+        'oppiaDebouncer',
+        function($scope, $attrs, $timeout, $element, LABEL_FOR_CLEARING_FOCUS,
+          oppiaDebouncer) {
           var guppyDivElt = $element[0].querySelector('.guppy-div');
+
+          /**
+           * Adds a button overlay and invisible text field used to bring up
+           * the keyboard on mobile devices.
+           *
+           * TODO: On iOS/Safari, keyboard may only appear on first press.
+           * This may not be a significant issue as the MathExpressionInput is
+           * recreated if the given answer is incorrect.
+           */
+          var addMobileButtonOverlay = function() {
+            /**
+             * Checks if the guppy div has a width and height greater than 0,
+             * if not schedules a timeout to run again after 100ms. If the
+             * guppy div has a valid width/height, position the invisible
+             * button directly over the guppy div. For mobile browsers, focus()
+             * can only be called from within an onclick handler. Using a form
+             * element was more reliable than attaching the handler to the
+             * guppy div directly.
+             */
+            var positionButtonOverlay = function() {
+              var guppyOffset = $(guppyDivElt).position();
+              var guppySize = document.querySelector('.guppy-div')
+                .getBoundingClientRect();
+              console.log(guppySize, $('#startMathInputButton'));
+
+              // If the guppy div hasn't rendered yet, retry after 100ms.
+              if (guppySize.width === 0 || guppySize.height === 0) {
+                $timeout(positionButtonOverlay, 100);
+              } else {
+                $('#startMathInputButton').css({
+                  top: guppyOffset.top,
+                  left: guppyOffset.left,
+                  width: guppySize.width,
+                  height: guppySize.height
+                });
+              }
+            };
+            positionButtonOverlay();
+
+            // The focus() call must be in a click event handler and on a text
+            // field to make the mobile keyboard appear.
+            $('#startMathInputButton').click(function() {
+              guppyInstance.activate();
+
+              var fakeInputElement = document.
+                querySelector('#fakeInputForMathExpression');
+              fakeInputElement.focus();
+
+              // Place the cursor at the end of the text input, so that the
+              // user can use backspace to delete.
+              fakeInputElement.setSelectionRange(fakeInputElement.value.length,
+                fakeInputElement.value.length);
+            });
+
+            // Mapping characters back to Mousetrap codes.
```

Comment:
Guppy uses the Mousetrap [(ccampbell/mousetrap)](https://github.com/ccampbell/mousetrap) library for keyboard input. Mousetrap has a function call which allows sending key events, but for symbols like "^" they need to be sent as "shift+6" to be recognized. This object is used to convert symbols into strings that Mousetrap accepts via the Mousetrap.trigger() function.

Would that be a better comment? I'm not sure how it should be explained.